src/textual/app.py

```diff
@@ -76,8 +78,11 @@ class ActionError(Exception):
     pass
 
 
+ReturnType = TypeVar("ReturnType")
+
+
 @rich.repr.auto
-class App(DOMNode):
+class App(Generic[ReturnType], DOMNode):
```

Comment:
This allows you to specify the return type on `run()` So if you want to return an int you might do this:

```python
class MyApp(App[int]):
```

Reply:
Came from a Textual issue I can't seem to find now. But a user wan't to be able to launch TUIs within a larger app to get info from the user. The data could be an int, str, or something more complex.